app/main.py

```diff
@@ -1,8 +1,35 @@
+
+
 class Person:
-    # write your code here
-    pass
 
+    people: dict[str] = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
+
+    def add_wife(self, wife: [str]) -> None:
+        if wife is not None:
+            self.wife = self.people[wife]
+
+    def add_husband(self, husband: [str]) -> None:
+        if husband is not None:
+            self.husband = self.people[husband]
+
+
+def create_person_list(
+        people: list[dict[str, str | int | None]]
+) -> list[Person]:
+    person_list = [
+        Person(person["name"], person["age"])
+        for person in people
+    ]
+
+    for person, properties in zip(person_list, people):
+        if properties.get("wife") is not None:
+            person.add_wife(properties["wife"])
```

Comment:
Really, you do not need a method for this action. Because you have the same checking in this method(for None). Better to do in this line: `person.wife = self.people[properties["wife"]]` and you can remove method `add_wife`
The same with husband

Reply:
As for me, it the most simple way